### PR TITLE
Enable logging with intervals to save memory

### PR DESCRIPTION
**Pull request by nicofirst1**
Add option to allow for interval logging strategy, to save interactions every X batches.

## Description

Proposed changes include:
- Add **IntervalLoggingStrategy** (extends **LoggingStrategy**) that accepts an optional param **log_interval**. Every **log_interval** the strategy returns the interaction otherwise Interaction.empty().
- Modified **from_iterable** to filter out empty interactions

## Related Issue (if any)

Similar [pull request](https://github.com/facebookresearch/EGG/pull/230)

## Motivation and Context

When training with limited memory resources, appending all the interactions to a list may  cause OOM errors. Allowing  the logging strategy to save interactions every X steps (otherwise empty) gives more flexibility.

### CODE REVIEW

**Comment by nicofirst1** [comment]
@robertodessi What is going on with the wandb errors? I haven't imported anything new.

**Comment by robertodessi** [comment]
maybe a conflict in wandb versions between the CI server and EGG? 
I don't have access to this repo anymore, nothing much I can do about it, sorry!

**Comment by nicofirst1** [comment]
Oh, I see. Thank you for the tip!

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,13 +7,13 @@ jobs:
   build:
     working_directory: ~/EGG
     docker:
-      - image: circleci/python:3.6
+      - image: cimg/python:3.12
```
Comment (robertodessi) [inline .circleci/config.yml:10]: EGG was developed and tested with python ~3.6 maybe, why testing with 3.12?

Comment (nicofirst1) [reply .circleci/config.yml:10]: Isn't it time to update to a more recent version? Moreover, the docker circle image has been [deprecated](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034/9)

**Comment by robertodessi** [comment]
@nicofirst1 CI has just been deleted for this repo (look at the latest commit), it makes sense given it's not actively supported anymore

**Comment by nicofirst1** [comment]
I saw... such a pity. I guess I'll use my own fork

**Pull request closed without merging**